CMakeLists.txt

```diff
@@ -93,8 +99,14 @@ endif ()
 
 set (OPENCL_ICD_LOADER_HEADERS_DIR "${CMAKE_CURRENT_SOURCE_DIR}/inc" CACHE PATH "Path to OpenCL Headers")
 
-add_library (OpenCL ${OPENCL_ICD_LOADER_SOURCES})
+if (${OPENCL_ICD_LOADER_BUILD_SHARED_LIBS} OR ${BUILD_SHARED_LIBS})
```

Comment:
Can you please check that this does what you want it to do?

Since we have already incorporated the state of `BUILD_SHARED_LIBS` into the default state for `OPENCL_ICD_LOADER_BUILD_SHARED_LIBS` above I think we might just want this to be:

```cmake
if (${OPENCL_ICD_LOADER_BUILD_SHARED_LIBS})
```

Reply:
Right, this can be removed now that the suggestion from Mathias is in place